lib/util/batch.js

```diff
@@ -121,7 +121,25 @@ function supportsCql31(version) {
   return version.major > 3 || (version.major === 3 && version.minor >= 1);
 }
 
-var endsWithSemicolonRegex = new RegExp(/\s*;\s*$/);
+function containsDeleteOrUpdateStatement(query) {
+  var match = false;
+  if (query instanceof Batch) {
+    _.each(query.context.queries, function (subQuery) {
+      if (match) { return; }
+      if (containsDeleteOrUpdateStatement(subQuery)) {
+        match = true;
+      }
+    });
+  } else {
+    if (containsDeleteOrUpdateRegex.test(query.context.cql)) {
+      match = true;
+    }
+  }
+  return match;
+}
+
+var containsDeleteOrUpdateRegex = /(?:update|delete)\s+[^\s,\(\)]/i;
```

Comment:
This `RegExp` (as [most `RegExp` are](https://blog.liftsecurity.io/2014/11/03/regular-expression-dos-and-node.js?utm_source=nodeweekly&utm_medium=email)) is vulnerable to DOS attacks . Not a big deal since any consumer of `priam` should be sanitizing input before passing it a query.